### PR TITLE
[ticket/13891] Added reparser:list and reparser:reparse to CLI

### PR DESCRIPTION
**Pull request by s9e**
https://tracker.phpbb.com/browse/PHPBB3-13891

- [x] Depends on https://tracker.phpbb.com/browse/PHPBB3-13803 / #3579

### CODE REVIEW

File: phpBB/phpbb/console/command/reparser/list_all.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	* Constructor
+	*
+	* @param \phpbb\user $user
+	* @param ContainerBuilder $container Container used to locate the reparsers
+	*/
+	public function __construct(\phpbb\user $user, ContainerBuilder $container)
+	{
+		parent::__construct($user);
+		$this->reparser_names = array();
+		foreach (array_keys($container->findTaggedServiceIds('text_reparser.plugin')) as $name)
```
Comment (Nicofuma) [inline phpBB/phpbb/console/command/reparser/list_all.php]: use a \phpbb\di\service_collection instead

**Comment by Nicofuma** [comment]
beside my comment, +1

**Comment by s9e** [comment]
@Nicofuma I replaced the collection, added a progress bar and added a couple of options in case someone wants to test it on a smaller range of posts. AFAIK it should be good to go.

```bash
./phpBB/bin/phpbbcli.php reparser:reparse post_text \
	-vvv              \
	--ansi            \
	--range-min  123  \
	--range-max  4567 \
	--range-size 50
```

**Comment by Nicofuma** [comment]
looks great, just for the design I think it could be great to use the symfony style

for the style: https://github.com/phpbb/phpbb/pull/3514/files#diff-2b8f0912fa7a05395a2997d0e1949a25R98

for the progress bar: https://github.com/phpbb/phpbb/pull/3514/files#diff-5f4a6cdd29f5a7c27bdd0b98356cfef4R190 ($this->io is an instance of a Symfony Style)

and with a nice success/error message at the end: https://github.com/phpbb/phpbb/pull/3514/files#diff-5f4a6cdd29f5a7c27bdd0b98356cfef4R175

**Comment by s9e** [comment]
@Nicofuma I don't know when I'll be able to look into that. I'm busy investigating memory usage/chasing down possible memory leaks.

@CHItA Is there a way for you to move the CLI styling to the `phpbb\console` namespace and make it reusable?

**Comment by Nicofuma** [comment]
Actually the CI styling is already in Symfony. In this file there is just a few tweaks for the installer.

I'll send you a PR to improve the design.

**Comment by s9e** [comment]
Awesome, thanks! :+1:

**Comment by s9e** [comment]
!set WIP

**Comment by Nicofuma** [comment]
why WIP?

**Comment by s9e** [comment]
@Nicofuma How do you feel about showing the elapsed/estimated time and memory usage in the progress bar?

**Comment by s9e** [comment]
!unset WIP

**Comment by Nicofuma** [comment]
I like it but why there is 2 times?

**Pull request merged**